Formula/irrlicht.rb

```diff
@@ -26,27 +25,36 @@ class Irrlicht < Formula
 
   depends_on xcode: :build
 
+  depends_on "jpeg"
+  depends_on "libpng"
+
+  uses_from_macos "bzip2"
+  uses_from_macos "zlib"
+
   on_linux do
     depends_on "libx11"
     depends_on "libxxf86vm"
     depends_on "mesa"
   end
 
+  # Use libraries from Homebrew or macOS
+  patch do
+    url "https://github.com/Homebrew/formula-patches/raw/69ad57d16cdd4ecdf2dfa50e9ce751b082d78cf9/irrlicht/use-system-libs.patch"
+    sha256 "70d2534506e0e34279c3e9d8eff4b72052cb2e78a63d13ce0bc60999cbdb411b"
+  end
+
+  # Update Xcode project to use libraries from Homebrew and macOS
+  patch do
+    url "https://github.com/Homebrew/formula-patches/raw/69ad57d16cdd4ecdf2dfa50e9ce751b082d78cf9/irrlicht/xcode.patch"
+    sha256 "2cfcc34236469fcdb24b6a77489272dfa0a159c98f63513781245f3ef5c941c0"
+  end
```

Comment:
Anything here we can upstream?

Reply:
OTOH:

- I'm not familiar enough with Xcode to know if it's possible to do conditional compilation. More research needed. If it's possible, also need to see if macros can be defined via command-line on Linux.
- Irrlicht's jpeg is outdated but it might not be anymore on trunk. More research needed.